pkg/controller/validationscache.go

```diff
@@ -46,15 +49,20 @@ func newValidationResource(
 
 type validationCache map[validationKey]validationResource
 
+// newValidationCache returns a new empty instance of validationCache struct
 func newValidationCache() *validationCache {
 	return &validationCache{}
 }
 
+// has returns a boolean if the given key exist in the instance
 func (vc *validationCache) has(key validationKey) bool {
-	_, ok := (*vc)[key]
-	return ok
+	_, exists := (*vc)[key]
+	return exists
 }
 
+// store creates a key-value pair on the current instance
+// it uses given object to create a validationKey and a validationResource structs
+// constraint: if a key already exists, it will overwrite the value
```

Comment:
```suggestion
// store caches a 'ValidationOutcome' for the given 'Object'.
// constraint: cached outcomes will be updated in-place for a given object and
// consecutive updates will not preserve previous state.
```

Better not to reveal internal details to consumers of the API through documentation.